src/main/java/com/groupseven/sysc4806project/ShoppingCart.java

```diff
@@ -44,7 +44,29 @@ public void setBooks(List<Book> books) {
         this.books = books;
     }
 
-    public void checkout() {
+    public void addBook(Book book){
+        if (book != null){
+            this.books.add(book);
+        }
+    }
 
+    public void removeBook(int id){
+        this.books.removeIf(book -> book.getId() == id);
+    }
+
+    public Book getBook(int id){
+        for (Book book: books){
+            if (book.getId() == id){
+                return book;
+            }
+        }
+        return null;
+    }
+
+    public void clearCart(){
```

Comment:
Being picky about method names but we already know this item is a cart, I think `clearCart()` is a bit redundant. `clear()` seems good enough